kubernetes/kubernetes.go

```diff
@@ -228,6 +228,24 @@ func (in *K8SClient) IsGatewayAPI() bool {
 	return *in.isGatewayAPI
 }
 
+// Is IstioAPI checks whether Istio API is installed or not
+func (in *K8SClient) IsIstioAPI() bool {
+	if in.Istio() == nil {
+		return false
+	}
+	if in.isIstioAPI == nil {
+		isIstioAPI := false
+		_, err := in.k8s.Discovery().RESTClient().Get().AbsPath("/apis/networking.istio.io").Do(in.ctx).Raw()
+		if err == nil {
+			isIstioAPI = true
+		} else if !errors.IsNotFound(err) {
+			log.Warningf("Error checking Istio API configuration: %v", err)
+		}
+		in.isIstioAPI = &isIstioAPI
+	}
```

Comment:
This is how the other checks are done too so it's fine to keep it this way but I wonder if it'd be better to have finer grained checks per api group/resource rather than just checking a single group and assuming the rest are present.

Reply:
I would continue doing this at least for now, maybe in a follow up issue we can change this to a more granular design (including the gateway api too). For multicluster I think this is enough. What do you think?.